api/filters.py

```diff
@@ -23,3 +23,34 @@ def filter_with_multiple_query_values(self, queryset, name, value):
     class Meta:
         model = Asset
         fields = ['asset_type', 'model_number', 'email']
+
+
+class UserFilter(filters.FilterSet):
+    cohort = filters.CharFilter(
+        field_name='cohort',
+        lookup_expr='iexact',)
+
+    email = filters.CharFilter(
+        field_name='email',
+        lookup_expr='istartswith',
+        method='filter_by_email_or_allocated_asset_count',)
+
+    asset_count = filters.CharFilter(
+        field_name='allocated_asset_count',
+        lookup_expr='iexact',
+        method='filter_by_email_or_allocated_asset_count',)
```

Comment:
Add a [label](https://django-filter.readthedocs.io/en/latest/ref/filters.html#label) argument to the `asset_count` field with a value of '_exact allocated asset count_'.